src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor.cs

```diff
@@ -13,41 +12,35 @@ namespace Microsoft.CodeAnalysis.SymbolDisplay
 {
     internal abstract partial class AbstractSymbolDisplayVisitor : SymbolVisitor
     {
-        protected readonly ArrayBuilder<SymbolDisplayPart> builder;
-        protected readonly SymbolDisplayFormat format;
-        protected readonly bool isFirstSymbolVisited;
-        protected readonly bool inNamespaceOrType;
+        private ArrayBuilder<SymbolDisplayPart> _builder;
+        private SymbolDisplayFormat _format;
+        private bool _isFirstSymbolVisited;
+        private bool _inNamespaceOrType;
 
-        protected readonly SemanticModel semanticModelOpt;
-        protected readonly int positionOpt;
+        private SemanticModel? _semanticModelOpt;
+        private int _positionOpt;
 
-        private AbstractSymbolDisplayVisitor _lazyNotFirstVisitor;
-        private AbstractSymbolDisplayVisitor _lazyNotFirstVisitorNamespaceOrType;
+        private AbstractSymbolDisplayVisitor? _lazyNotFirstVisitor;
+        private AbstractSymbolDisplayVisitor? _lazyNotFirstVisitorNamespaceOrType;
 
-        protected AbstractSymbolDisplayVisitor(
-            ArrayBuilder<SymbolDisplayPart> builder,
-            SymbolDisplayFormat format,
-            bool isFirstSymbolVisited,
-            SemanticModel semanticModelOpt,
-            int positionOpt,
-            bool inNamespaceOrType = false)
+        protected AbstractSymbolDisplayVisitor()
         {
-            Debug.Assert(format != null);
+            // These values will be set to non-null by 'Allocate' when an instance is obtained from the pool
+            _builder = null!;
+            _format = null!;
+        }
```

Comment:
nit: remove and just make into initializers on the fields.

Reply:
⏱️ Will fix

Reply:
➡️ Fixed